### PR TITLE
Improve TractionAux

### PR DESCRIPTION
**Pull request by arovinelli**
allows to output traction in global coordinates using the "X,Y,Z" flags.
It also allows computing the first PK traction (collinear to the Cauchy traction but acting on the undeformed area)

### CODE REVIEW

**Comment by arovinelli** [comment]
the tests seem to be stuck, however this a piece for checking the correctness of LDCZM
@reverendbedford

**Comment by reverendbedford** [comment]
Also missing the gold file.

File: src/auxkernels/TractionAux.C [hunk 1]
```diff
@@ -17,45 +17,76 @@
...
 MooseEnum TractionAux::traction_component() {
-  return MooseEnum("normal shear1 shear2 shear_norm");
+  return MooseEnum("normal shear1 shear2 shear_norm X Y Z");
 }
 
 InputParameters TractionAux::validParams() {
   InputParameters params = MaterialAuxBase<RankTwoTensor>::validParams();
   params.addRequiredParam<MooseEnum>("scalar_type",
                                      TractionAux::traction_component(),
                                      "Type of scalar output");
```
Comment (reverendbedford) [inline src/auxkernels/TractionAux.C:27]: You should be more descriptive.  Include the possible options in the docstring.

File: src/auxkernels/TractionAux.C [hunk 2]
```diff
@@ -17,45 +17,76 @@
...
 
 TractionAux::TractionAux(const InputParameters &parameters)
     : MaterialAuxBase<RankTwoTensor>(parameters), _normals(_assembly.normals()),
-      _scalar_type(getParam<MooseEnum>("scalar_type")) {
+      _scalar_type(getParam<MooseEnum>("scalar_type")),
+      _PK1(getParam<bool>("PK1")),
+      _large_kinematics(getParam<bool>("large_kinematics")),
+      _ndisp(coupledComponents("displacements")) {
   if (!this->boundaryRestricted())
     mooseError("TractionAux Kernle must be boundary restricted ");
```
Comment (reverendbedford) [inline src/auxkernels/TractionAux.C:46]: Spelling.

File: src/auxkernels/TractionAux.C [hunk 3]
```diff
@@ -17,45 +17,76 @@
...
+                                                   (*_grad_disp[1])[_qp],
+                                                   (*_grad_disp[2])[_qp]));
+    RankTwoTensor R;
+    F.getRUDecompositionRotation(R);
+    normal = R * _normals[_qp];
+    if (_PK1)
+      dadA = F.det() * (F.inverse().transpose() * _normals[_qp]).norm();
+  }
 
   // compute rotation matrix required to aligne the normal with the 1,0,0 vector
```
Comment (reverendbedford) [inline src/auxkernels/TractionAux.C:75]: Spelling

File: src/auxkernels/TractionAux.C [hunk 4]
```diff
@@ -17,45 +17,76 @@
...
   // compute the traction
   RealVectorValue traction = stress_R * local_normal;
+  // scale traction on original area
+  traction *= dadA;
 
   // extract components
-  Real val;
+  Real val = 0;
   switch (_scalar_type) {
   case 0:
```
Comment (reverendbedford) [inline src/auxkernels/TractionAux.C:91]: Your cases should be the enum values, not the integer equivalents, to make this easier to read.

**Pull request merged**